power_grid_model_c/power_grid_model_c/src/handle.hpp

```diff
@@ -24,4 +28,21 @@ struct PGM_Handle {
     [[no_unique_address]] power_grid_model::BatchParameter batch_parameter;
 };
 
+template <class Exception = std::exception, class Functor>
+auto call_with_catch(PGM_Handle* handle, Functor func, PGM_Idx error_code, std::string_view extra_msg = {})
+    -> std::invoke_result_t<Functor> {
+    if (handle) {
+        PGM_clear_error(handle);
+    }
+    using ReturnValueType = std::remove_cvref_t<std::invoke_result_t<Functor>>;
+    static std::conditional_t<std::is_void_v<ReturnValueType>, int, ReturnValueType> const empty{};
+    try {
+        return func();
+    } catch (Exception const& e) {
+        handle->err_code = error_code;
+        handle->err_msg = std::string(e.what()) + std::string(extra_msg);
+        return static_cast<ReturnValueType>(empty);
+    }
+}
+
 #endif
```

Comment:
```suggestion
#endif

```

Reply:
Can you enforce this in clang-format?

Reply:
no idea, i'll have a look

Reply:
requires `clang-format-16` https://releases.llvm.org/16.0.0/tools/clang/docs/ClangFormatStyleOptions.html#insertnewlineateof